lib/compile_to_file/source_file.rb

```diff
@@ -0,0 +1,48 @@
+module CompileToFile
+  class SourceFile
+
+    attr_reader :path
+    attr_reader :swaps
+    attr_reader :skip_requires
+
+    def initialize(path, skip_requires)
+      @path = path
+      @swaps = []
+      @skip_requires = skip_requires
+      yield self if block_given?
+    end
+
+    def swap
+      new_swap = CodeSwap.new
+      swaps << new_swap
+      new_swap
+    end
+
+    def processed_source
+      processed = ""
+
+      File.open(path, "r") do |code_file|
+
+        # Annotate with source file path for debug help
+        processed << "## #{path} ##\n"
+
+        code = code_file.read
+
+        swaps.each { |s| s.apply_to!(code, path) }
+
+        # Strip comments and shebangs
+        code.gsub!(/^\s*#.*\n/, "")
```

Comment:
Is this stripping required for correctness? I understand the require probably will be, but not as much the comments, other than optimizing for size; and we could package the code as a gziped here doc if we needed to deal with that.